renderer.php

```diff
@@ -320,22 +326,21 @@ public function valid($records) {
         foreach ((array)$records as $key => $value) {
             $tmpparam = str_replace(" ", "_", $value->name);
             $tmpparam = optional_param($tmpparam, '', PARAM_RAW);
-
-            if ($value->required == "Yes" && $value->type != "HTML") {
-                if ($value->type == "Email" && (stripos($tmpparam, "@") === false ||
+            if ($value->required == "Yes" && $value->type != get_string('html', 'local_pages')) {
+                if ($value->type == get_string('email', 'local_pages') && (stripos($tmpparam, "@") === false ||
                         stripos($tmpparam, ".") === false)
                 ) {
-                    $this->error_fields[$value->name] = "Please Supply a valid email address for " . $value->name;
+                    $this->error_fields[$value->name] = get_string('validemail', 'local_pages', ['email' => $value->name ]);
```

Comment:
Because the lang string has changed from $a->prop to $a, no need to pass associative array just the value

```suggestion
                    $this->error_fields[$value->name] = get_string('validemail', 'local_pages', $value->name);
```

Reply:
Thanks - added